common/djangoapps/student/models.py

```diff
@@ -1718,6 +1721,10 @@ def upgrade_deadline(self):
             # Replicate that behavior here by returning None if the personalized deadline is in the past.
             if datetime.now(UTC) >= self.dynamic_upgrade_deadline:
                 return None
+
+            if self.course_overview.get_course_mode(CourseMode.VERIFIED) is None:
```

Comment:
Before this change we were returning valid upgrade deadlines for courses that did not actually have a verified track.